### PR TITLE
Update only disp params if already present in the file

### PR DESCRIPTION
**Pull request by morcuended**
Trying to fix #543 for real data since there are no disp related parameters in those files. They are only in MC files

I removed the re-calculation of `r` because it is already done by the `hillas_parameters` function.

### CODE REVIEW

**Review by rlopezcoto — APPROVED** [review]
thanks @morcuended 
maybe let's get this merged and solve the problem generated by #543 and later decide to go and write some unit tests for real data

**Comment by morcuended** [comment]
For an MC file:
image

File: lstchain/scripts/lstchain_dl1ab.py [hunk 1]
```diff
@@ -158,19 +152,29 @@ def main():
...
-                                                                          dl1_container['y'].to_value(u.m),
-                                                                          params['src_x'][ii],
-                                                                          params['src_y'][ii],
-                                                                          )
-                dl1_container['disp_dx'] = disp_dx
-                dl1_container['disp_dy'] = disp_dy
-                dl1_container['disp_norm'] = disp_norm
-                dl1_container['disp_angle'] = disp_angle
-                dl1_container['disp_sign'] = disp_sign
+                if ['disp_dx', 'disp_dy', 'disp_norm', 'disp_angle', 'disp_sign'] in dl1_params_input:
```
Comment (morcuended) [inline lstchain/scripts/lstchain_dl1ab.py:157]: This is not working as expected, either I check that only one of those `disp_` parameters are present or do something like `if disp_params.issubset(dl1_params_input)` which slows a lot the runtime. Any suggestion?

Comment (maxnoe) [reply lstchain/scripts/lstchain_dl1ab.py:157]: This is a check that has to run once per file. How can that be slowing it down by a lot?

```
if set(dl1_params.intersection({'disp_dx', 'disp_dy', 'disp_norm', 'disp_angle', 'disp_sign'}):
```

Comment (maxnoe) [reply lstchain/scripts/lstchain_dl1ab.py:157]: Also:

```

In [5]: %timeit set(dl1_columns).intersection({'disp_dx', 'disp_dy', 'disp_norm', 'disp_angle', 'disp_sign'})
777 ns ± 2.24 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

how can this be a problem?

Comment (maxnoe) [reply lstchain/scripts/lstchain_dl1ab.py:157]: Dependening on if you want `all` behaviour or `any` behaviour (i would go for `any`), use `intersection` or `issubset`

Comment (morcuended) [reply lstchain/scripts/lstchain_dl1ab.py:157]: I was doing that inside the loop over all the events `for ii, row in enumerate(image_table)`

Comment (maxnoe) [reply lstchain/scripts/lstchain_dl1ab.py:157]: Even then. It takes less than *a microsecond*`. But why do you do that in the loop? The `parameters_to_update` will be fixed for the whole file, or not?

Comment (morcuended) [reply lstchain/scripts/lstchain_dl1ab.py:157]: Yes, you are right. I will refactor that.

I got the impression in my first tests that it was much slower but then tested with the another `if 'disp_dx'...` condition and it was pretty much the same.

Comment (morcuended) [reply lstchain/scripts/lstchain_dl1ab.py]: Done

**Review by rlopezcoto — APPROVED** [review]

File: lstchain/scripts/lstchain_dl1ab.py [hunk 2]
```diff
@@ -158,19 +153,21 @@ def main():
                     length = np.rad2deg(np.arctan2(dl1_container.length, foclen))
                     dl1_container.width = width
                     dl1_container.length = length
-                    dl1_container.r = np.sqrt(dl1_container.x ** 2 + dl1_container.y ** 2)
```
Comment (vuillaut) [inline lstchain/scripts/lstchain_dl1ab.py:161]: Sorry, why did you remove that line?

Comment (vuillaut) [reply lstchain/scripts/lstchain_dl1ab.py:161]: nevermind, got it

**Review by vuillaut — APPROVED** [review]

File: lstchain/scripts/lstchain_dl1ab.py (hunk 2 above)
Comment (morcuended) [reply lstchain/scripts/lstchain_dl1ab.py:161]: It is already calculated by `hillas_parameters` function some lines above if I'm not wrong.

```
return HillasParametersContainer(
        x=u.Quantity(cog_x, unit),
        y=u.Quantity(cog_y, unit),
        r=u.Quantity(cog_r, unit),
        phi=Angle(cog_phi, unit=u.rad),
        intensity=size,
        length=u.Quantity(length, unit),
        length_uncertainty=u.Quantity(length_uncertainty, unit),
        width=u.Quantity(width, unit),
        width_uncertainty=u.Quantity(width_uncertainty, unit),
        psi=Angle(psi, unit=u.rad),
        skewness=skewness_long,
        kurtosis=kurtosis_long,
    )
```
